<?xml version="1.0"?>
<launch>
<arg name="paused" default="false" doc="Starts gazebo in paused mode" />
  <arg name="gui" default="true" doc="Starts gazebo gui" />

  <!-- Start up Gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch" >
    <arg name="paused" value="false"/>
    <arg name="gui" value="$(arg gui)"/>
  </include>

  <!-- Spawn Teeterbot with specific settings -->
  <include file="$(find teeterbot_gazebo)/launch/teeterbot_old.launch" >
    <!-- Name and spawn pose -->
    <arg name="robot_name" value="teeterbot" />
    <arg name="start_x" value="0.0" />
    <arg name="start_y" value="0.0" />
    <arg name="start_z" value="0.2" />
    <arg name="start_yaw" value="0.0" />

    <!-- Physical properties -->
    <!--arg name="body_length" value="0.8" />
    <arg name="body_width" value="0.3" />
    <arg name="body_depth" value="0.3" />
    <arg name="body_mass" value="10.0" />
    <arg name="wheel_mass" value="1.0" />
    <arg name="wheel_radius" value="0.2" />
    <arg name="training_wheels" value="false" /-->

    <!-- Simulation behavior settings -->
    <!--arg name="pub_ground_truth" value="false" />
    <arg name="auto_reset_orientation" value="false" />
    <arg name="auto_reset_delay" value="2.0" /-->

    <!-- Control mode (set only one) -->
    <!--arg name="voltage_mode" value="false" />
    <arg name="torque_mode" value="true" />
    <arg name="speed_mode" value="false" /-->
  </include>

  <!-- <node pkg="teeterbot_gazebo" name="teeterbot_gazebo_angle" type="get_angle.py" output="screen"> 
  </node>

  <node pkg="teeterbot_gazebo" name="teeterbot_pid_controller" type="pid_controller.py" output="screen"> 
  </node>
 -->
<!-- 
  <node pkg="teeterbot_gazebo" name="sb_LQR" type="sb_LQR.py" output="screen"> 
  </node> -->

</launch>
